spring-integration-core/src/main/java/org/springframework/integration/support/serializer/VersionTolerantJavaDeserializer.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.serializer;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.springframework.core.NestedIOException;
+import org.springframework.core.serializer.Deserializer;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+/**
+ * A {@link Deserializer} implementation that reads an input stream using Java serialization in a
+ * {@link VersionTolerantObjectInputStream serialVersionUID tolerant} manner.
+ *
+ * <pre>
+ * <code>
+ * 	VersionTolerantJavaDeserializer deser = new VersionTolerantJavaDeserializer();
```

Comment:
@artembilan this is the workaround we discussed. 

Because the `serialVersionUID` is verified very early in the `ObjectInputStream` (namely before readObject on the actual class) the only technique to allow us to inspect and decide on that UID is by overriding [readClassDescriptor](https://github.com/spring-projects/spring-integration/pull/3738/files#diff-4dccaf3dfbd729f0f6c2cdde477a3e9440eab9539c23903498e65d620bf620f6R43). 

This makes this more of an intrusive scenario as we have to figure out how to wedge in the custom OIS. We do not want to do this in the framework. We do, however want to give users (@german22) a workaround to use if they so choose until 6.0.0 comes out where we will make sure all serializables have UID fields (breaking change though). 

@artembilan I am not sure what the delivery vehicle for this would be. We can chat more about this tomorrow. 

The workaround is thoroughly tested as I did not want to provide this w/o knowing it was solid. Usage is described below. The idea being one would have to hook in the custom OIS into whatever store they are using that needs the version tolerant deserialization.